src/Impostor.Server/Plugins/PluginLoader.cs

```diff
@@ -155,5 +167,103 @@ private static void CheckPaths(IEnumerable<string> paths)
                 assemblyInfos.Add(new AssemblyInformation(assemblyName, path, isPlugin));
             }
         }
+
+        private static List<PluginInformation> LoadOrderPlugins(IEnumerable<PluginInformation> plugins)
+        {
+            var pluginDict = new Dictionary<string, PluginInformation>();
+            var hardDependencies = new Dictionary<string, List<string>>();
+
+            foreach (var plugin in plugins)
+            {
+                pluginDict[plugin.Name] = plugin;
+                hardDependencies[plugin.Name] = plugin
+                    .Dependencies
+                    .Where(p => p.DependencyType == DependencyType.HardDependency)
+                    .Select(p => p.Name)
+                    .ToList();
+            }
+
+            var presentPlugins = pluginDict.Keys.ToList();
+
+            // Check whether the Hard Dependencies are present and remove those without.
+            var checkedPlugins = CheckHardDependencies(presentPlugins, hardDependencies);
+
+            var dependencyGraph = checkedPlugins.ToDictionary(p => p, _ => new List<string>());
+
+            foreach (var plugin in checkedPlugins)
+            {
+                foreach (var dependency in pluginDict[plugin].Dependencies.Where(d => checkedPlugins.Contains(d.Name)))
+                {
+                    if (dependency.DependencyType == DependencyType.LoadBefore)
+                    {
+                        dependencyGraph[dependency.Name].Add(plugin);
+                    }
+                    else
+                    {
+                        dependencyGraph[plugin].Add(dependency.Name);
+                    }
+                }
+            }
```

Comment:
Any checks when there is a circular dependency?

Reply:
No, right now if there is a circular dependency the server will arbitrarily load either one of the plugins first. A circularity check could be added although I'm not sure whether it is really needed since if there exists a circular dependency the plugin is just not written correctly.